src/core/net/SRP_CODER.md

```diff
@@ -0,0 +1,309 @@
+# SRP Coder
+
+This document specifies SRP coder used to encode an SRP message into a compact, compressed format, reducing the message size. On the server, the received coded message can be decoded to reconstruct an exact replica of the original message.
+
+The SRP protocol uses DNS resource record format to convey information about registered services and host addresses. The use of these records and adherence to the DNS Update general format introduces overhead and can lead to larger SRP message sizes, which is undesirable for Thread network. The SRP coder aims to optimize the message, resulting in much smaller messages containing all the same information reducing overall protocol overhead.
+
+## Compact Integer Format
+
+- Works with unsigned integer of different length (`uint16`, `uint32`)
+- Number is encoded as one or more segment(s).
+- Segments are one byte (8-bit) long except for the first segment which may have fewer bits (2-7 bits).
+- First (MSB) bit in each segment is the “continuation bit” indicating whether there are more segments to follow (`1`) or if it is the last segment (`0`).
+- The remaining bits after the MSB provide the numerical `uint` bit values in big-endian order.
+
```

Comment:
This had me scratching my head for a while. It looks like the idea is that the rightmost byte contains seven bits of the integer, and each byte to the left contains another seven bits, until we get to the leftmost byte, which contains up to seven bits. It might save head-scratching for the next person if you say that explicitly. I think there isn't another way to do it, so they will eventually figure it out, so not a huge problem.

Reply:
Updated this section to add more details and examples. There is a subtle aspect regarding the first segment, which can have a smaller bit-length.